PopulateTools/gobierto
# WIP GobiertoData add dcat catalog endpoint

Closes #2671


## :v: What does this PR do?

add endpoint `/api/v1/data/datasets/catalog.xml` which show an rdf/xml (catalog in dcat format) 



## :mag: How should this be manually tested?
0. configure new site options:
```
bin/rails c
site = Site.find_by(domain: "*********")
GobiertoSeeds::GobiertoData::Recipe.run(site) if site.configuration.gobierto_data_enabled?
```
check configuration was loaded in selected site: `backoffice > customize site`

![Screenshot from 2021-05-07 07-13-02](https://user-images.githubusercontent.com/288355/117400882-cd5b2f80-af03-11eb-8e05-7988d5e1df53.png)



1. create datasets with specific features:
 - a dataset with license
 - a dataset without license
 - a dataset with source url
 - a dataset without source url
 - a dataset with category
 - a dataset without category
  
2. visit endpoint api/v1/data/datasets/catalog.xml
3. copy xml payload
4. paste payload into [https://www.w3.org/RDF/Validator/](https://www.w3.org/RDF/Validator/)
- check valid passed (rdf validation)
5. paste payload into
 [FOR DCAT](http://www.dcat.be/validator/)
 [FOR DCAT-AP](https://www.itb.ec.europa.eu/shacl/dcat-ap/upload)

- check valid passed (dcat validation) 
- you can ignore messages like this:
```
The property: title can't have multiple objects
The property: description can't have multiple objects
```
